### ISSUE TITLE
Bugs related to Business Contact (section) in bizregisteration

### ISSUE THREAD

**Issue by priyankshukla** [OrderMail/blacklabel#30]
There are following bugs encountered in business registration screen:
a) A empty business contact gets added in order to click on +(plus) button under business contact section.
b) It's throwing required (contact and email address fields) error in business contact section while hitting 'Register' button. However I already added a business contact and other mandatory fields have been entered.
c) It's allowing me to remove the primary contact (business), I meant, it is registering a business without having a primary business contact.

It is the reference of #27

**Comment by vikasagarwal** [comment]
Fixed 1, 2, and 3.

**Comment by vikasagarwal** [comment]
@priyankshukla , @amit-k86 Please review. I'm moving labels.

**Comment by priyankshukla** [comment]
Here are the steps I performed this morning:
1. All the users and associated business documents have been removed from mongodb.
2. I registered a new user in blacklabel successfully (without selecting any business).
3. I clicked the Blacklabel (logo) in top left. It redirects me on dashboard. Following two issues needs to be highlight:
a) He/she should not redirect to dashboard without login. I am going to raise another bug ref #4 
b) User is not associated with any business, he/she should redirect to bizregistration page rather than dashboard (I verified from backend that User has business id as 'null' in Users document)

Suggestion:
To redirect to dashboard, we need to check, whether user is activated or not, apart from checking the user login & business association functionality.

Apart from above, I would like to highlight that grunt is failing while executing jshint:all

**Comment by priyankshukla** [comment]
Ah my bad.... The above mentioned issues are either bug or new requirement. That I will log separably to reference with appropriate story.

I am landed to bizregistration screen after successfully logged-in. 
Testing now...

**Comment by priyankshukla** [comment]
Tested! Here are my test result:
a) I am able to add the space & special character as customer title under Business Contacts section.
b) I am able to add the space in email address under Business Contacts section.
c) I added two extra blank business contacts (with space) apart from default business contact (logged-in user details). 
Here is the scene, User has 3 business contact rows: 
1 with default logged-in user details and the primary flag set on in this.
2 appears blank business contact, this I saved with spaces (customer title and email address).
3 appears blank business contact, this I saved with spaces (customer title and email address).
 When I deleted the last business contact (space one) the primary flag switch to second business contact (space one) from default business contact, I have not played with primary flag though.

d) There is no success message after bizregistration successfully.
Also, I would like to get confirm whether it is appropriate to delete the default user contact (who logged-in and creating business). Off-course we can set the primary flag to other business contact.
